microlith/loki/alerting/couchbase/couchbase-rules.yaml

```diff
@@ -200,3 +200,41 @@ groups:
           title: "Malformed vBucket ({{ $labels.vbucket }}) in bucket: {{ $labels.bucket}}, on node: {{ $labels.hostname }}"
           description: "Files in vbucket, {{ $labels.vbucket }}, have been corrupted"
           remediation: "Review your infrastructure for signs of disk problems or any other issues. Alternatively, navigate to your Data storage directory and enter the following command, making sure to fill in the correct data: `/opt/couchbase/bin/couch_dbck --verbose --json <couchbase_server_data_directory>/<bucket>/<malformed_vb_file>`, then take the response and provide it to Couchbase Technical Support."
+
+      - alert: CB90056-connectionsRejected-Warning
+        expr: |
+          count_over_time({file="memcached.log"} |= "Too many open connections" | json message="message" | line_format "{{.message}}" | pattern "<_>WARNING Too many open connections. Current/Limit for port 11210: <current_limit>; total: <total> " [24h]) == 1
+        for: 1m
+        labels:
+          job: couchbase_fluent_bit
+          kind: node
+          severity: warning
+          health_check_id: CB90056
+          health_check_name: connectionsRejected
+          cluster: '{{ $labels.cluster }}'
+          node: '{{ $labels.hostname }}'
+          current_limit: '{{ $labels.current_limit }}'
+          total: '{{ $labels.total }}'
+        annotations:
+          title: "Too Many Open Connections to Memcached (Node: {{ $labels.hostname }})"
+          description: "Memcached has reached the maximum number of open connections ({{ $labels.current_limit }}), this means it will start rejecting new connections"
+          remediation: Contact your network/application team to see if there are any applications keeping open a large number of connections.
+
+      - alert: CB90056-connectionsRejected-Alert
+        expr: |
+          count_over_time({file="memcached.log"} |= "Too many open connections" | json message="message" | line_format "{{.message}}" | pattern "<_>WARNING Too many open connections. Current/Limit for port 11210: <current_limit>; total: <total> " [24h]) > 1
```

Comment:
I might be missing something really obvious, but these two rules look the same to me?

Reply:
One is ==1 the other >1 for warn and alert

Reply:
So the difference between the two is that we've seen more than one message per 24 hours, if I'm understanding that right? Perhaps that merits a comment to explain it, because it took me a while to realise that (and perhaps documenting / adding to the annotations too, because the user will be wondering why it's suddenly an alert as opposed to a warning).